From b43d73336a9209717adb038ca1fe3a808d3ce36e Mon Sep 17 00:00:00 2001
From: Jason Zaman <jason@perfinion.com>
Date: Tue, 30 May 2023 09:10:03 -0700
Subject: [PATCH 05/13] systemlib: Updates for Abseil 20220623 LTS

These targets are header-only and just need stub bazel targets
---
 third_party/absl/system.absl.functional.BUILD | 22 +++++++++++++++++++
 third_party/absl/system.absl.random.BUILD     | 12 ++++++++++
 2 files changed, 34 insertions(+)

diff --git a/third_party/absl/system.absl.functional.BUILD b/third_party/absl/system.absl.functional.BUILD
index a4f70acf35c..579181dec07 100644
--- a/third_party/absl/system.absl.functional.BUILD
+++ b/third_party/absl/system.absl.functional.BUILD
@@ -2,10 +2,32 @@ load("@rules_cc//cc:defs.bzl", "cc_library")
 
 package(default_visibility = ["//visibility:public"])
 
+cc_library(
+    name = "any_invocable",
+    deps = [
+        "//absl/base:base_internal",
+        "//absl/base:config",
+        "//absl/base:core_headers",
+        "//absl/meta:type_traits",
+        "//absl/utility",
+    ],
+)
+
 cc_library(
     name = "bind_front",
+    deps = [
+        "//absl/base:base_internal",
+        "//absl/container:compressed_tuple",
+        "//absl/meta:type_traits",
+        "//absl/utility",
+    ],
 )
 
 cc_library(
     name = "function_ref",
+    deps = [
+        "//absl/base:base_internal",
+        "//absl/base:core_headers",
+        "//absl/meta:type_traits",
+    ],
 )
diff --git a/third_party/absl/system.absl.random.BUILD b/third_party/absl/system.absl.random.BUILD
index 948de07751a..5ebd656be8e 100644
--- a/third_party/absl/system.absl.random.BUILD
+++ b/third_party/absl/system.absl.random.BUILD
@@ -51,3 +51,15 @@ cc_library(
         "//absl/types:span",
     ],
 )
+
+cc_library(
+    name = "bit_gen_ref",
+    deps = [
+        ":random",
+        "//absl/base:core_headers",
+        "//absl/base:fast_type_id",
+        "//absl/meta:type_traits",
+        "//absl/random/internal:distribution_caller",
+        "//absl/random/internal:fast_uniform_bits",
+    ],
+)
-- 
2.41.0

